multipaz/src/commonMain/kotlin/org/multipaz/mdoc/zkp/ZkSystemRepository.kt

```diff
@@ -0,0 +1,84 @@
+package org.multipaz.mdoc.zkp
+
+import kotlinx.io.bytestring.ByteString
+
+/**
+ * Singleton object that manages registration and lookup of [ZkSystem] implementations,
+ * and provides central interface for proof generation and verification.
+ */
+object ZkSystemRepository {
```

Comment:
Using a singleton like this is problematic and makes unit testing more difficult and also makes it more difficult for big applications where separate parts might use this. It's also not what we're doing for `DocumentTypeRepository`, `SecureAreaRepository`, and elsewhere. So please change this to be a class.